src/Amazon.Lambda.Tools/LambdaPackager.cs

```diff
@@ -92,7 +99,7 @@ private static bool IsAmazonLinux2(IToolLogger logger)
             else if (string.IsNullOrWhiteSpace(containerImageForBuild))
             {
                 // Use a default container image if Use Container is set to true, or if we need to build NativeAOT on non-AL2
-                if ((useContainerForBuild.HasValue && useContainerForBuild.Value) || (isNativeAot && !IsAmazonLinux2(logger)))
+                if ((useContainerForBuild.HasValue && useContainerForBuild.Value) || (isNativeAot && !IsAmazonLinux(logger)))
```

Comment:
I think previously we had the benefit of only having AL2 to deploy to, so we knew what OS we had to build on. But here, what if the customer is using an AL2023 build image (for example, a SAM CLI provided container image that is AL2023), but their Lambda is still configured to deploy to provided.al2? 

Seems like we should either check that the `OS build version = OS runtime version` or we could just say like `.net 6 = al2` and `.net 8 = al2023`. Locking a .NET version to an Amazon Linux version does reduce the testing and support landscape, but limits customer options.

Reply:
Or maybe AL2 and AL2023 are close enough that they are cross-build compatible?

Reply:
Seems like a rare issue that built on AL2 is deployed to AL2023 or vice versa, and it could be compatible anyway. No need to change it.